app/controllers/question_answer/votes_controller.rb

```diff
@@ -4,47 +4,47 @@ module QuestionAnswer
   class VotesController < ::ApplicationController
     before_action :ensure_logged_in
     before_action :find_vote_post
-    before_action :find_vote_user, only: [:create, :destroy]
+    before_action :ensure_can_see_post
```

Comment:
Covered up a security hole here. Previously anyone would have been able to vote on behalf of any user even if they do not have access to the post.